src/main/java/me/zeroeightsix/kami/util/CoordUtil.java

```diff
@@ -0,0 +1,104 @@
+package me.zeroeightsix.kami.util;
+
+import net.minecraft.client.Minecraft;
+
+import me.zeroeightsix.kami.util.Coord;
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Scanner;
+
+import static me.zeroeightsix.kami.util.MessageSendHelper.sendChatMessage;
+
+/**
+ * @author dominikaaaa
+ * Created by dominikaaaa on 18/02/20
+ * Updated by wnuke on 14/04/20 and 26/04/20
+ */
+public class CoordUtil {
+    public static final String coordsLogFilename = "KAMIBlueCoords.txt";
+    public static int[] getCurrentCoord(boolean chunk) {
+        Minecraft mc = Minecraft.getMinecraft();
+        int[] currentCoords = {(int) mc.player.posX, (int) mc.player.posY, (int) mc.player.posZ};
+        if (chunk) {
+            return new int[] {currentCoords[0]/16, currentCoords[1]/16, currentCoords[2]/16};
+        } else {
+            return currentCoords;
+        }
+    }
+
+    public static int[] writePlayerCoords(String locationName, boolean chunk) {
+        int[] coords = getCurrentCoord(chunk);
+        writeCoords(coords, locationName, Boolean.toString(chunk), coordsLogFilename);
+        return coords;
+    }
+
+    public static void writeCoords(int[] xyz, String locationName, String chunk, String filename) {
+        try {
+            FileWriter fW = new FileWriter(filename, true);
+            fW.write(formatter(xyz[0], xyz[1], xyz[2], locationName, chunk));
+            fW.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static String formatter(int x, int y, int z, String locationName, String chunk) {
+        SimpleDateFormat sdf = new SimpleDateFormat("HH:mm:ss dd/MM/yyyy");
+        String time = sdf.format(new Date());
+        return "x: " + x + ", y: " + y + ", z: " + z + ", chunk: " + chunk + ", name: \"" + locationName + "\", date: \"" + time + "\"\n";
+    }
+    public static Coord[] coordsLogToArray() {
+        Coord[] coordsArray = new Coord[lineCount(coordsLogFilename)];
+        try {
+            File coordsFile = new File(coordsLogFilename);
+            Scanner coordsReader = new Scanner(coordsFile);
+            int line = 0;
+            while (coordsReader.hasNextLine()) {
+                String coordsRaw = coordsReader.nextLine();
+                if (coordsRaw.length() > 0) {
+                    String[] split1 = coordsRaw.split("x: ")[1].split(", y: ");
```

Comment:
These `split` calls are a mess. Much better to use regex, or even better to use a defined format such as JSON/XML/Properties

Reply:
Change file format to json, using gson.